TestTimeout fails on mac

@hugoArregui told me this has been failing on mac. It is failing on my mac.
Now I understand what is going on and I believe it is a bug in the test.
The fix is incoming.

### Your environment.
  - Version: 14ec8
  - Browser: n/a

### What did you do?
Run unitest on mac with:
```
go test -v -run TestTimeout
```

### What did you expect?
The test to pass.

### What happened?
```
=== RUN   TestTimeout
--- FAIL: TestTimeout (30.00s)
    transport_test.go:43: Connection timed out early. (after 29900 ms)
```

--- a/README.md
+++ b/README.md
@@ -38,6 +38,7 @@ Check out the **[contributing wiki](https://github.com/pion/webrtc/wiki/Contribu
 * [Hugo Arregui](https://github.com/hugoArregui)
 * [Adam Kiss](https://github.com/masterada)
 * [Aleksandr Razumov](https://github.com/ernado)
+* [Yutaka Takeda](https://github.com/enobufs)
 
 ### License
 MIT License - see [LICENSE](LICENSE) for full text

--- a/transport_test.go
+++ b/transport_test.go
@@ -23,15 +23,18 @@ func TestStressDuplex(t *testing.T) {
 
 func testTimeout(t *testing.T, c *Conn, timeout time.Duration) {
 	const pollrate = 100 * time.Millisecond
+	const margin = 20 * time.Millisecond // allow 20msec error in time
 	statechan := make(chan ConnectionState)
 	ticker := time.NewTicker(pollrate)
 
+	startedAt := time.Now()
+
 	for cnt := time.Duration(0); cnt <= timeout+defaultTaskLoopInterval; cnt += pollrate {
 		<-ticker.C
+
 		err := c.agent.run(func(agent *Agent) {
 			statechan <- agent.connectionState
 		})
-
 		if err != nil {
 			// we should never get here.
 			panic(err)
@@ -39,9 +42,11 @@ func testTimeout(t *testing.T, c *Conn, timeout time.Duration) {
 
 		cs := <-statechan
 		if cs != ConnectionStateConnected {
-			if cnt < timeout {
-				t.Fatalf("Connection timed out early. (after %d ms)", cnt/time.Millisecond)
+			elapsed := time.Since(startedAt)
+			if elapsed+margin < timeout {
+				t.Fatalf("Connection timed out %f msec early", elapsed.Seconds()*1000)
 			} else {
+				t.Logf("Connection timed out in %f msec", elapsed.Seconds()*1000)
 				return
 			}
 		}

